api/src/main/java/jakarta/faces/component/UISelectOne.java

```diff
@@ -95,23 +95,24 @@ public VisitResult visit(VisitContext visitContext, UIComponent target)
                     {
                         // check if the is empty (see ) or if it's not valid (means this path has been taken already)
                         // See conditions listed under spec: uiselectone#processValidators
-                        if(isEmpty(submittedValue) && isSubmittedAlready(target)){
+                        if(isSubmittedAlready(target)){
                             previouslySubmittedOrValidated = true;
                             return VisitResult.COMPLETE;
                         }          
                     }
                     return VisitResult.ACCEPT;
                 }
             });
-        }
 
-        if(previouslySubmittedOrValidated){
-            // Skip further validation due to either 
-            // 1) one of the submissions are not valid (for instance, required, but none submitted (see jakarta/faces#329))
-            // 2) submitted value has been found and validated
-            return;
+            if(previouslySubmittedOrValidated){
+                // Skip further validation due to either 
+                // 1) one of the submissions are not valid (for instance, required, but none submitted (see jakarta/faces#329))
+                // 2) submitted value has been found and validated
+                return;
+            }
```

Comment:
Did some refactoring here as we should really only return when `isEmpty(submittedValue)` is true since we're just trying to skip further validation.